source/common/http/http3/conn_pool.cc

```diff
@@ -65,10 +65,13 @@ allocateConnPool(Event::Dispatcher& dispatcher, Random::RandomGenerator& random_
       host, priority, dispatcher, options, transport_socket_options, random_generator, state,
       [&quic_stat_names,
        &scope](HttpConnPoolImplBase* pool) -> ::Envoy::ConnectionPool::ActiveClientPtr {
+        pool->addLog(Logger::Logger::debug, "Create Http/3 client");
         // If there's no ssl context, the secrets are not loaded. Fast-fail by returning null.
         auto factory = &pool->host()->transportSocketFactory();
         ASSERT(dynamic_cast<Quic::QuicClientTransportSocketFactory*>(factory) != nullptr);
         if (static_cast<Quic::QuicClientTransportSocketFactory*>(factory)->sslCtx() == nullptr) {
+          pool->addLog(Logger::Logger::warn, "Fail to create Http/3 client. Transport socket "
```

Comment:
So does ENVOY_LOG not work because this lambda is being executed outside of an object that inherits from Loggable? Is that why we need the addLog() methods? I wonder if there is some clever envoy idiom for logging in lambdas?

Reply:
Yes, I saw listener_manage_impl also use similar trick, so I guess this is the way...

Reply:
"This is the way" - The Mandalorian

:)

Reply:
I found another approach! Change the lambda to take `spd::logger&` as another argument and use ENVOY_LOG_TO_LOGGER. Fortunately there isn't too many places needs interface change.

Reply:
Oh, nice! That's great!